go-controller/pkg/ovn/gress_policy.go

```diff
@@ -227,48 +227,65 @@ func (gp *gressPolicy) getMatchFromIPBlock(lportMatch, l4Match string) []string
 	return ipBlockMatches
 }
 
-// addNamespaceAddressSet adds a new namespace address set to the gress policy
-func (gp *gressPolicy) addNamespaceAddressSet(name, portGroupName string) {
+// addNamespaceAddressSet adds a namespace address set to the gress policy
+// ff it does not exist and returns `false` if it does.
+func (gp *gressPolicy) addNamespaceAddressSet(name string) bool {
 	v4HashName, v6HashName := addressset.MakeAddressSetHashNames(name)
 	v4HashName = "$" + v4HashName
 	v6HashName = "$" + v6HashName
 
 	if gp.peerV4AddressSets.Has(v4HashName) || gp.peerV6AddressSets.Has(v6HashName) {
-		return
+		return false
```

Comment:
Hang on, isn't this a bug? If we go from single-stack to dual-stack, then is this condition incorrect?

Reply:
Aha, nevermind, we restart when changing mode.